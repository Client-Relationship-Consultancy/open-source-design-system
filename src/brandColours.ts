export const colourPalette = {
  examplePalette: {
    primary: {
      main: {
        hex: "#A42824",
        RGB: "164,40,36",
        on: "#FFFFFF",
      },
      light: {
        hex: "#DA342F",
        RGB: "218,52,47",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#661614",
        RGB: "102,22,20",
        on: "#FFFFFF",
      },
    },
    secondary: {
      main: {
        hex: "#0B466B",
        RGB: "11,70,107",
        on: "#FFFFFF",
      },
      light: {
        hex: "#599CAF",
        RGB: "89,156,175",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#05263B",
        RGB: "5,38,59",
        on: "#FFFFFF",
      },
    },
    complimentary: {
      main: {
        hex: "#FFEE52",
        RGB: "255,238,82",
        on: "#272727",
      },
      light: {
        hex: "#FFF59D",
        RGB: "255,245,157",
        on: "#272727",
      },
      dark: {
        hex: "#CEB900",
        RGB: "206,185,0",
        on: "#272727",
      },
    },
    action: {
      main: {
        hex: "#743C8F",
        RGB: "116,60,143",
        on: "#FFFFFF",
      },
      light: {
        hex: "#AD65CF",
        RGB: "173,101,207",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#4E2662",
        RGB: "79,38,98",
        on: "#FFFFFF",
      },
    },
    error: {
      main: {
        hex: "#FF7320",
        RGB: "255,115,32",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#C54901",
        RGB: "197,73,1",
        on: "#FFFFFF",
      },
    },
    background: {
      hex: "#F6F6F6",
      RGB: "246,246,246",
    },
    surface: {
      hex: "#FFFFFF",
      RGB: "255,255,255",
      on: "#272727",
    },
    white: {
      hex: "#FFFFFF",
      RGB: "255,255,255",
      on: "#272727",
    },
    black: {
      main: { hex: "#272727", RGB: "39,39,39", on: "#FFFFFF" },
      tint80: { hex: "#696969", RGB: "105,105,105", on: "#FFFFFF" },
      tint60: { hex: "#A0A0A0", RGB: "160,160,160", on: "#FFFFFF" },
      tint40: { hex: "#C7C7C7", RGB: "199,199,199", on: "#272727" },
      tint20: { hex: "#ECECEC", RGB: "236,236,236", on: "#272727" },
    },
  },
};

interface IColourCodes {
  hex: string;
  RGB: string;
  on?: string;
}
interface IColourVariations {
  main: IColourCodes;
  light?: IColourCodes;
  dark?: IColourCodes;
}

interface IBlackVariations {
  main: IColourCodes;
  tint80: IColourCodes;
  tint60: IColourCodes;
  tint40: IColourCodes;
  tint20: IColourCodes;
}

export interface IColourPalette {
  primary: IColourVariations;
  secondary: IColourVariations;
  complimentary: IColourVariations;
  action: IColourVariations;
  error: IColourVariations;
  background: IColourCodes;
  surface: IColourCodes;
  white: IColourCodes;
  black: IBlackVariations;
}
