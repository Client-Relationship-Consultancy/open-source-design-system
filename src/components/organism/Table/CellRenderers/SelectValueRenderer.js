import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import SelectArrowIcon from "../../../../assets/select-arrow-icon.svg"

export const Wrapper = styled.div`
  overflow: hidden;
  text-overflow: ellipsis;
  width: calc(100% - 25px);
  :after {
    content: url("${SelectArrowIcon}");
    position: absolute;
    top: 0.55rem;
    right: 0.55rem;
    transition: opacity 0.15s;
    opacity: 0;
  }
  :hover, :focus {
    cursor: pointer;
    :after {
      opacity: 1;
    }
  }
`

Wrapper.displayName = "Wrapper"

class SelectValueRenderer extends React.Component {
  renderValue = () => {
    const fullOption =
      this.props.colDef.cellEditorParams &&
      this.props.colDef.cellEditorParams.values &&
      this.props.colDef.cellEditorParams.values.find(option => option.value === this.props.value)

    return fullOption ? fullOption.label : this.props.value
  }

  render = () => <Wrapper>{this.renderValue()}</Wrapper>
}

SelectValueRenderer.displayName = "SelectValueRenderer"

SelectValueRenderer.propTypes = {
  colDef: PropTypes.shape({
    cellEditorParams: PropTypes.shape({
      values: PropTypes.array,
    }),
  }),
  value: PropTypes.string,
}

export default SelectValueRenderer
