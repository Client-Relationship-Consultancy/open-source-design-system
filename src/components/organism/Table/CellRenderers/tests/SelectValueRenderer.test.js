import React from "react"
import { shallow, mount } from "enzyme"
import { ThemeProvider } from "styled-components"

import { colourPalette } from "../../../../../brandColours"
import SelectValueRenderer, { Wrapper } from "../SelectValueRenderer"

describe("Wrapper", () => {
  it("should match snapshot", () => {
    const component = mount(
      <ThemeProvider theme={colourPalette.examplePalette}>
        <Wrapper />
      </ThemeProvider>,
    )
    expect(component).toMatchSnapshot()
  })
})

describe("SelectValueRenderer", () => {
  const startEditingCellMock = jest.fn()
  const commonProps = {
    api: { startEditingCell: startEditingCellMock },
    value: "girls",
    colDef: {
      cellEditorParams: { values: ["alpha", "bravo", "charlie"] },
    },
  }
  it("should match snapshot", () => {
    const component = shallow(<SelectValueRenderer {...commonProps} />)
    expect(component).toMatchSnapshot()
  })

  it("should render the given value even if it is not in the options", () => {
    const component = shallow(<SelectValueRenderer {...commonProps} />)
    expect(component.instance().renderValue()).toEqual("girls")
  })

  it("should render a string with a space when given a falsy value to ensure the cursor is still a pointer on hover", () => {
    const component = shallow(<SelectValueRenderer {...commonProps} value={null} />)
    expect(component.instance().renderValue()).toEqual(" ")
  })

  it("should fire the startEditingCell function when the dropdown button is clicked a single time", () => {
    const component = shallow(<SelectValueRenderer {...commonProps} value={null} />)
    component.find("button").simulate("click")
    expect(startEditingCellMock).toBeCalledTimes(1)
  })

  it("should render the label of the given value from the options", () => {
    const values = [
      { label: "foo", value: "bar" },
      { label: "hello", value: "world" },
      { label: "no", value: "yes" },
      { label: "spice", value: "girls" },
    ]
    const component = shallow(
      <SelectValueRenderer {...commonProps} colDef={{ cellEditorParams: { values } }} />,
    )
    expect(component.instance().renderValue()).toEqual("spice")
  })

  it("should render the given value if options do not exist", () => {
    const component = shallow(
      <SelectValueRenderer {...commonProps} colDef={{ cellEditorParams: { values: [] } }} />,
    )
    expect(component.instance().renderValue()).toEqual("girls")
  })
})
