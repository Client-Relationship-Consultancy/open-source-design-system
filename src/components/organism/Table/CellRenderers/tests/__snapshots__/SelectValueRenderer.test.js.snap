// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectValueRenderer should match snapshot 1`] = `
<Wrapper>
  girls
</Wrapper>
`;

exports[`Wrapper should match snapshot 1`] = `
.c0 {
  white-space: pre;
  overflow: hidden;
  text-overflow: ellipsis;
  width: calc(100% - 20px);
}

.c0:after {
  content: url(".styledComponentId");
  position: absolute;
  top: 0.55rem;
  right: 0.55rem;
  -webkit-transition: opacity 0.15s;
  transition: opacity 0.15s;
  opacity: 0;
}

.c0:hover,
.c0:focus {
  cursor: pointer;
}

.c0:hover:after,
.c0:focus:after {
  opacity: 1;
}

<Wrapper>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "SelectValueRenderer__Wrapper-sc-1ccrjb7-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "white-space:pre;overflow:hidden;text-overflow:ellipsis;width:calc(100% - 20px);:after{content:url(\\"",
            ".styledComponentId",
            "\\");position:absolute;top:0.55rem;right:0.55rem;transition:opacity 0.15s;opacity:0;}:hover,:focus{cursor:pointer;:after{opacity:1;}}",
          ],
        },
        "displayName": "Wrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "SelectValueRenderer__Wrapper-sc-1ccrjb7-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</Wrapper>
`;
